src/App.tsx

```diff
@@ -1,61 +1,139 @@
 import './App.scss';
+import { useState } from 'react';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+
+import { User } from './types/User';
+import { Todo } from './types/Todo';
+import { TodoList } from './components/TodoList';
+
+function findUserById(userId: number): User | null {
+  const foundUser = usersFromServer.find(user => user.id === userId);
+
+  return foundUser || null;
+}
+
+export const todoWithUser: Todo[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: findUserById(todo.userId),
+}));
+
+function findUserByName(userName: string): User | null {
+  return usersFromServer.find(user => user.name === userName) || null;
+}
 
 export const App = () => {
+  const [todos, setTodos] = useState(todoWithUser);
+  const [title, setTitle] = useState('');
+  const [selectedUser, setSelectedUser] = useState('');
+  const [isTitleError, setIsTitleError] = useState(false);
+  const [isUserNameError, setIsUserNameError] = useState(false);
+
+  const handleSubmit = (event: React.ChangeEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    if (selectedUser && title) {
+      setTodos(todosPrev => {
+        const maxTodoId = Math.max(...todosPrev.map(todo => todo.id));
+        const newUser = findUserByName(selectedUser);
```

Comment:
What u will do if users will have the same name? Let's save user id in state

Reply:
Fixed.